File: swift-sdk/Sources/Cleanster/Api/BlacklistApi.swift
import Foundation

/// API methods for blocking specific cleaners from being assigned to bookings.
public final class BlacklistApi {
    private let client: CleansterClient
    init(client: CleansterClient) { self.client = client }

    /// List all blacklisted cleaners.
    public func listBlacklistedCleaners() async throws -> ApiResponse<AnyCodable> {
        return try await client.requestRaw(method: "GET", path: "/v1/blacklist/cleaner")
    }

    /// Add a cleaner to the blacklist.
    ///
    /// - Parameters:
    ///   - cleanerId: The cleaner's user ID.
    ///   - reason: Optional internal note explaining the reason.
    public func addToBlacklist(cleanerId: Int, reason: String? = nil) async throws -> ApiResponse<AnyCodable> {
        let body = BlacklistRequest(cleanerId: cleanerId, reason: reason)
        return try await client.requestRaw(method: "POST", path: "/v1/blacklist/cleaner", body: body)
    }

    /// Remove a cleaner from the blacklist.
    public func removeFromBlacklist(cleanerId: Int) async throws -> ApiResponse<AnyCodable> {
        let body = BlacklistRequest(cleanerId: cleanerId)
        return try await client.requestRaw(method: "DELETE", path: "/v1/blacklist/cleaner", body: body)
    }
}

File: kotlin-sdk/src/main/kotlin/com/cleanster/api/BlacklistApi.kt
package com.cleanster.api

import com.cleanster.CleansterClient
import com.cleanster.model.*

/** API methods for blocking specific cleaners from being assigned to bookings. */
class BlacklistApi internal constructor(private val client: CleansterClient) {

    /** List all blacklisted cleaners. */
    suspend fun listBlacklistedCleaners(): ApiResponse<List<Any>> = client.request(
        method = "GET",
        path   = "/v1/blacklist/cleaner",
    )

    /**
     * Add a cleaner to the blacklist.
     *
     * @param cleanerId The cleaner's user ID.
     * @param reason    Optional internal note.
     */
    suspend fun addToBlacklist(cleanerId: Int, reason: String? = null): ApiResponse<Map<String, Any>> = client.request(
        method = "POST",
        path   = "/v1/blacklist/cleaner",
        body   = BlacklistRequest(cleanerId = cleanerId, reason = reason),
    )

    /** Remove a cleaner from the blacklist. */
    suspend fun removeFromBlacklist(cleanerId: Int): ApiResponse<Map<String, Any>> = client.request(
        method = "DELETE",
        path   = "/v1/blacklist/cleaner",
        body   = BlacklistRequest(cleanerId = cleanerId),
    )
}
